### ISSUE TITLE
[java] JUnitTestsShouldIncludeAssert: Add support for Hamcrest MatcherAssert.assertThat

### ISSUE THREAD

**Issue by TheBoegl** [pmd/pmd#2288]
**Affects PMD Version:** 6.20.0
**Rule:** [JUnitTestsShouldIncludeAssert](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#junittestsshouldincludeassert)
**Description:**
the `MatcherAssert.assertThat` is not counted as assertion rule

**Code Sample demonstrating the issue:**

```java
import org.hamcrest.MatcherAssert;
import org.hamcrest.Matchers;
import org.junit.jupiter.api.Test;

class JUnitTestsShouldIncludeAssertRule {
    @Test
    void hamcrest_assert_should_pass_include_test() {
        MatcherAssert.assertThat("java/rule/bestpractices/JUnitTestsShouldIncludeAssertRule should pass for MatcherAssert.assertThat", "PMD 6.20.0", Matchers.is("PMD 6.20.0"));
    }
}
```

**Running PMD through:** *Gradle 6.1.1*

**Comment by adangel** [comment]
It makes sense, to enhance the rule to support hamcrest matchers as well. We currently support the JUnit asserts, aspectj soft asserts, Mockito calls.

In any case, we should update the rule description to clearly say, which assertion frameworks are recognized.
